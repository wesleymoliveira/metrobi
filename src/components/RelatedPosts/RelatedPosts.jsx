import { Typography } from '../Typography/Typography';
import * as S from './RelatedPosts.styled';

export function RelatedPosts() {
	return (
		<S.Wrapper data-testid='related-posts'>
			<Typography size='xxlarge'>
				Related posts
			</Typography>
		</S.Wrapper>
	);
}
