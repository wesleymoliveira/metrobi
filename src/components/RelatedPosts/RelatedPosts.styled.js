import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.relatedPostsBackground};
    width: 20%;
    height: 12rem;
    display: flex;
    justify-content: center;
    align-items: start;
    text-align: center;
    overflow: hidden;
  `}
`;
