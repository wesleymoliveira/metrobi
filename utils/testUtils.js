import React from "react";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import theme from "../src/styles/theme";

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options });
};

const snapshotRenderer = (component) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
};

export * from "@testing-library/react";

export { customRender as render, snapshotRenderer  };
