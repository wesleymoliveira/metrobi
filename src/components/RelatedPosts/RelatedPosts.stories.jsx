import React from 'react';
import { RelatedPosts } from './RelatedPosts';

const RelatedPostsArgTypes = {
  label: {
    name: 'label',
    type: { name: 'string', required: false },
    description: 'overwritten description',
    table: {
      type: { 
        summary: 'something short', 
        detail: 'something really really long' 
      },

    },
    control: {
      type: 'string'
    }
  }
};

export default {
	title: 'components/RelatedPosts',
	component: RelatedPosts,
	argTypes: RelatedPostsArgTypes
};

export const Default = (args) => <RelatedPosts {...args} />;
