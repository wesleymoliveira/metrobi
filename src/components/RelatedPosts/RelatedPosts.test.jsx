import { screen } from '@testing-library/react';

import { RelatedPosts } from './RelatedPosts';
import { render, snapshotRenderer } from '../../../utils/testUtils';

describe('<RelatedPosts />', () => {
		it('should render the component', () => {
		render(<RelatedPosts />);
		
		expect(screen.getByTestId(/related-posts/i)).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const tree = snapshotRenderer(<RelatedPosts />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should match the original background-color', () => {
		const tree = snapshotRenderer(<RelatedPosts />).toJSON();
		//make sure the background-color is the original one
		expect(tree).toHaveStyleRule('background-color', '#F1CCDE')
	});

});
