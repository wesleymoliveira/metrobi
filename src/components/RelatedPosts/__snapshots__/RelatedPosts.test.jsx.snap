// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RelatedPosts /> should match snapshot 1`] = `
.c1 {
  color: #FAFAFA;
  font-size: 2.8rem;
}

.c0 {
  background-color: #F1CCDE;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  text-align: center;
  overflow: hidden;
}

<div
  className="c0"
  data-testid="related-posts"
>
  <h1
    className="c1"
    color="white"
    data-testid="typography"
    size="xxlarge"
  >
    Related posts
  </h1>
</div>
`;
